packages/angular/ssr/src/app-engine.ts

```diff
@@ -120,6 +130,23 @@ export class AngularAppEngine implements AngularServerAppManager {
 
     return entryPoints.get(potentialLocale);
   }
+
+  /**
+   * Retrieves the HTTP headers for a given request based on the URL pathname
+   * of statically generated (SSG) pages.
+   *
+   * @param request - The incoming HTTP request object.
+   * @returns A record of HTTP headers or `undefined` if there are no headers for the given pathname.
+   */
+  getHeaders(request: Request): Readonly<Record<string, string>> | undefined {
+    if (this.manifest.staticPathsHeaders.size === 0) {
+      return undefined;
+    }
+
+    const { pathname } = stripIndexHtmlFromURL(new URL(request.url));
+
+    return this.manifest.staticPathsHeaders.get(pathname);
```

Comment:
Should we consider always returning a map? That can help avoid the need to make extra checks in apps code.

```suggestion
  getHeaders(request: Request): Readonly<Record<string, string>> {
    if (this.manifest.staticPathsHeaders.size === 0) {
      return new Map();
    }

    const { pathname } = stripIndexHtmlFromURL(new URL(request.url));

    return this.manifest.staticPathsHeaders.get(pathname) ?? new Map();
```